[team4_pair2] 자동차 경주 게임 2단계 구현 및 테스트 추가

Kelicia91 opened the pull request:
step 2 코드리뷰 부탁드립니다.

Commit:
자동차 경주 게임 2단계 구현 및 테스트 추가

javajigi commented on src/main/java/racingcar/Car.java:
```diff
@@ -1,10 +1,19 @@
     private int position;
 
     public Car() {
+        this("");
```
기본 생성자가 왜 필요한가?

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -2,18 +2,20 @@
 public class RacingGame {
 
     private List<Car> cars;
+    private Map<Integer, List<String>> rankMap;
```
cars와 중복 값을 가지는 인스턴스 변수이다.
rankMap 인스턴스 변수 없이 구현해본다.

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -2,18 +2,20 @@
-            cars.add(new Car());
+        rankMap = new HashMap<>();
+
+        String[] names = StringUtility.splitWithComma(input);
```
입력 값에 대한 예외 처리는 어디서 하는 것이 좋을까?

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -34,9 +36,38 @@ public int getRandomNo() {
+    public List<String> getWinnerNameList() {
+        int maxPosition = 0;
+        for (Car car : cars) {
+            maxPosition = Math.max(maxPosition, car.getPosition());
```
상태를 가지는 객체의 데이터를 꺼내려(get)하지 말고 객체에 메시지를 보내라는 원칙에 따라 구현한다.
car.getPosition() 처럼 데이터를 get하지 않고 구현해 본다.

javajigi commented on src/test/java/utility/StringUtilityTest.java:
```diff
@@ -2,11 +2,29 @@
+    public void joinWithComma() {
+        String actual = StringUtility.joinWithComma(Arrays.asList("a", "b"));
+        assertEquals("a, b", actual);
+    }
```
우승자를 구하는 부분이 복잡한 로직으로 보여지는데 테스트하는 부분이 없다.
이 부분에 대한 단위 테스트 코드를 구현해 본다.

javajigi commented on src/test/java/utility/StringUtilityTest.java:
```diff
@@ -2,11 +2,29 @@
+        String actual = StringUtility.joinWithComma(Arrays.asList("a", "b"));
+        assertEquals("a, b", actual);
+    }
 }
```
사용자에게 값을 입력받고 결과를 출력하는 main() 메소드가 없네요.
main() 메소드를 구현해서 동작 가능한 프로그램으로 만들어 보세요.

javajigi reviewed (changes requested):
프로그램을 실행하는 main() 메소드가 없네요.

Commit:
* refactoring step2
- 불필요한 Car 생성자 삭제
- rankMap 삭제
- 자동차 입력값 예외처리
- car.getPosition() 삭제
- main 메소드 추가
